src/structures/Embed.ts

```diff
@@ -1,25 +1,126 @@
+        return this.description = description
+    }
+
+    public setTimestamp(timestamp: Boolean | null) {
+        if(typeof timestamp === null) {
+            return this.description = null
+        }
+        if(typeof timestamp !== "boolean") throw new Error(`Expected type boolean but got ${typeof timestamp} for embed timestamp`)
+        return this.timestamp = Date.now()
+    }
+
+    public setColor(color: String | null) {
+        if(typeof color === null) {
+            return this.color = null
+        }
+        if(typeof color !== "string") throw new Error(`Expected type string but got ${typeof color} for embed color`)
+        return this.color = color //unfinished needs a way to set the correct format
+    }
+
+    public setFooter(description: String | null) { //unfinished
+        if(typeof description === null) {
+            return this.description = null
+        }
+        if(typeof description !== "string") throw new Error(`Expected type string but got ${typeof description} for embed description`)
+        return this.description = description
+    }
+
+    public setImage(description: String | null) { //unfinished
+        if(typeof description === null) {
+            return this.description = null
+        }
+        if(typeof description !== "string") throw new Error(`Expected type string but got ${typeof description} for embed description`)
+        return this.description = description
+    }
+
+    public setThumbnail(description: String | null) { //unfinished
+        if(typeof description === null) {
+            return this.description = null
+        }
+        if(typeof description !== "string") throw new Error(`Expected type string but got ${typeof description} for embed description`)
+        return this.description = description
+    }
+
+    public setVideo(description: String | null) { //unfinished
+        if(typeof description === null) {
+            return this.description = null
+        }
+        if(typeof description !== "string") throw new Error(`Expected type string but got ${typeof description} for embed description`)
+        return this.description = description
+    }
+
+    public setProvider(description: String | null) { //unfinished
+        if(typeof description === null) {
+            return this.description = null
+        }
+        if(typeof description !== "string") throw new Error(`Expected type string but got ${typeof description} for embed description`)
+        return this.description = description
+    }
+
+    public setAuthor(description: String | null) { //unfinished
+        if(typeof description === null) {
+            return this.description = null
+        }
+        if(typeof description !== "string") throw new Error(`Expected type string but got ${typeof description} for embed description`)
+        return this.description = description
+    }
+
+    public addField(description: String | null) { //unfinished
```

Comment:
might wanna make this addFields and have multiple fields, also not description

Reply:
> might wanna make this addFields and have multiple fields

Why?